### PR TITLE
[Docs] add missing `RFC` variants to URL functions

### LINKED ISSUE #1942: Document `cutToFirstSignificantXYZ` functions.
Create documentation for the `cutToFirstSignificantXYZ` functions. 

- [ ] cutToFirstSignificantSubdomainCustomRFC
- [ ] cutToFirstSignificantSubdomainCustomWithWWWRFC
- [x] cutToFirstSignificantSubdomainRFC
- [x] cutToFirstSignificantSubdomainWithWWWRFC

See #1833 for details on how to complete this issue.

Child issue of #1833.

### LINKED ISSUE #1953: Document `firstXYZ` functions.
Create documentation for the `firstXYZ` functions:

- [ ] firstSignificantSubdomainCustomRFC
- [ ] firstSignificantSubdomainRFC
- [ ] first_value_respect_nulls

See #1833 for details on how to complete this issue.

Child issue of #1833.

### LINKED ISSUE #1995: Document `portRFC` function.
Create documentation for the `portRFC` function. See #1833 for details on how to complete this issue.

Child issue of #1833.

### LINKED ISSUE #2024: Document `topLevelDomainRFC` function.
Create documentation for the `topLevelDomainRFC` function. See #1833 for details on how to complete this issue.

Child issue of #1833.

### PR DESCRIPTION
**Pull request by Blargian**
Updates URL functions page as part of the [missing functions project](https://github.com/ClickHouse/clickhouse-docs/issues/1833).

Closes [#2024](https://github.com/ClickHouse/clickhouse-docs/issues/2024)
Closes [#1995](https://github.com/ClickHouse/clickhouse-docs/issues/1995)
Closes [#1942](https://github.com/ClickHouse/clickhouse-docs/issues/1942)
Closes [#1953](https://github.com/ClickHouse/clickhouse-docs/issues/1953)

Adds
- `topLevelDomainRFC`
- `cutToFirstSignificantSubdomainCustomRFC`
- `cutToFirstSignificantSubdomainRFC`
- `domainRFC`
- `firstSignificantSubdomainCustomRFC`
- `firstSignificantSubdomainRFC`
- `cutToFirstSignificantSubdomainCustomWithWWWRFC`
- `cutToFirstSignificantSubdomainWithWWWRFC` 

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 899bb86f9a9ac00dbc55cc65c3f150e97950f58e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64494/899bb86f9a9ac00dbc55cc65c3f150e97950f58e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/sql-reference/functions/url-functions.md [hunk 1]
```diff
@@ -78,30 +78,78 @@ topLevelDomain(url)
...
+
 ``` text
 ┌─topLevelDomain('svn+ssh://www.some.svn-hosting.com:80/repo/trunk')─┐
 │ com                                                                │
 └────────────────────────────────────────────────────────────────────┘
 ```
 
+### topLevelDomainRFC
+
+Extracts the the top-level domain from a URL. It is similar to [topLevelDomain](#topleveldomain), but conforms to RFC 3986.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/url-functions.md:113]: Side note and not blocking this PR:

The beginning of the document currently contains sentence "All these functions do not follow the RFC.". The average reader will be confused by this.

Someone reading the docs will not really understand how `topLevelDomainRFC` differs from `topLevelDomain`. What is generally missing in the entire md file is the specific non-RFC3986 URL format (with examples) recognized by all functions documented in this file, and likewise the RFC3986 format recognized by the `*RFC` functions described in this file. [Here](https://github.com/ClickHouse/ClickHouse/pull/42274) is the PR which added the RFC variant, perhaps the contained examples are helpful to come up with better documentation (--> other PR).

**Review by rschu1ze — CHANGES_REQUESTED** [review]

File: docs/en/sql-reference/functions/url-functions.md (hunk 1 above)
Comment (Blargian) [reply docs/en/sql-reference/functions/url-functions.md:177]: @rschu1ze I've added the other missing `RFC` variants with some examples for which the RFC variant will work but not the non-`RFC` one. I've also made some changes to the beginning of the document to better clarify the difference. Let me know if we need to expand on this further or modify, remove something.

Comment (rschu1ze) [reply docs/en/sql-reference/functions/url-functions.md]: Should be good, thanks.

I also went once over the docs and pushed some fixups.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**